tests/Doctrine/Tests/ORM/Functional/ResultCacheTest.php

```diff
@@ -133,6 +149,65 @@ public function testUseResultCacheParams()
         $this->assertEquals($sqlCount + 2, count($this->_sqlLoggerStack->queries), "The next two sql should have been cached, but were not.");
     }
 
+    public function testEnableResultCache()
+    {
+        $cache = new ArrayCache();
+        $query = $this->_em->createQuery('select ux from Doctrine\Tests\Models\CMS\CmsUser ux');
+
+        $query->enableResultCache();
+        $query->setResultCacheDriver($cache);
+        $query->setResultCacheId('testing_result_cache_id');
+        $users = $query->getResult();
+
+        $this->assertTrue($cache->contains('testing_result_cache_id'));
+
+        $this->_em->getConfiguration()->setResultCacheImpl(new ArrayCache());
+    }
+
+    /**
+     * @group DDC-1026
+     */
+    public function testEnableResultCacheParams()
+    {
+        $cache    = new ArrayCache();
+        $sqlCount = count($this->_sqlLoggerStack->queries);
+        $query    = $this->_em->createQuery('select ux from Doctrine\Tests\Models\CMS\CmsUser ux WHERE ux.id = ?1');
+
+        $query->setParameter(1, 1);
+        $query->setResultCacheDriver($cache);
+        $query->enableResultCache();
+        $query->getResult();
+
+        $query->setParameter(1, 2);
+        $query->getResult();
+
+        $this->assertCount($sqlCount + 2, $this->_sqlLoggerStack->queries, "Two non-cached queries.");
+
+        $query->setParameter(1, 1);
+        $query->enableResultCache();
+        $query->getResult();
+
+        $query->setParameter(1, 2);
+        $query->getResult();
+
+        $this->assertCount($sqlCount + 2, $this->_sqlLoggerStack->queries, "The next two sql should have been cached, but were not.");
```

Comment:
- The next two sql what? queries?
- Why "next"? There is nothing after this line of code…

Reply:
@greg0ire These tests were copied from original test for `useResultCache()` method

Reply:
Ok, I took more time to read and I understand now: there are 2 queries, then a first assertion, and then 2 queries, so "next" means "after the 2 first queries" (it's the next set of 2 queries). I think you should add the missing word:

```suggestion
        $this->assertCount(
            $sqlCount + 2,
            $this->_sqlLoggerStack->queries,
            "The next two sql queries should have been cached, but were not."
        );
```

Another thing I don't understand about this test (and yours) is why `useResultCache(true)` and `enableResultCache()` are called twice on the same object. Do you maybe know?

Reply:
@greg0ire as long as removing the line with the second `enableResultCache()` (or `useResultCache(true)` does not change the behavior of the test, apparently it is just a mistake made by original test writer during copy-paste. I'd also reposition the first `$query->setParameter()` call to be closer to `$query->getResult()` as it is probably sematically more correct. Will do both yours and mine suggested changes in the next commit.